### PR TITLE
Updates for new `aorsf` engine for `rand_forest()`

### LINKED ISSUE #827: add `aorsf` engine to `rand_forest`?
Hello! 

Following the contribution guidelines, I am filing an issue to check in with the tidymodels team and ask if it would be alright for me to open a `parsnip` pull request that will add documentation for the `aorsf` engine to the `rand_forest` model. This engine would only be applicable to censored regression at the moment.

## background

In a [separate pull request](https://github.com/tidymodels/censored/issues/172) to `censored`, @hfrick has reviewed changes, helped me improve `aorsf`'s API, and requested I initiate a pull request to `parsnip`.

### PR DESCRIPTION
**Pull request by bcjaeger**
Closes #827

Hello!

I would like to include the oblique random survival forest in `parsnip` as an engine for censored regression. I have reviewed the guidelines for contributing engine level documentation and have made .Rmd and .R files for `aorsf` that mimic existing files for random forest engines, e.g., `partykit`.

Following the engine-specific guidelines in `inst/README-DOCS.md`, 

> 1. Use roxygen comments to generate the main .Rd file that R's help system uses.
> 2. Create an .Rmd file that has the details of the engine.
> 3. Knit the .Rmd file to .md so the .Rd file from step 1 can link it.
> 4. Once you are done, run devtools::document() (or use the RStudio IDE) to build the help files.

I had a little trouble on step 3. Knitting my .Rmd file, the error `package 'parsnip' required by 'censored' could not be found` occurred.  So I did not commit the .Rd file. I've tried to figure out what is causing `parsnip` to be undetected, but haven't figured it out yet. 

> If you make a PR with a new engine, you can work on the .Rmd file and the tidymodels folks can go through the process of merging it in (but let us know if you want us to do that in the PR). 

Since `aorsf` is a new engine and I am having trouble creating the .Rd file, may I request help getting the documentation files to knit as intended?

### CODE REVIEW

File: man/rmd/rand_forest_aorsf.Rmd [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+`r descr_models("rand_forest", "aorsf")`
+
+## Tuning Parameters
+
+```{r aorsf-param-info, echo = FALSE}
+defaults <-
+  tibble::tibble(parsnip = c("trees", "min_n", "mtry"),
+                 default = c("500L", "5L", "ceiling(sqrt(n_predictors))"))
```
Comment (bcjaeger) [inline man/rmd/rand_forest_aorsf.Rmd:12]: the use of ceiling(sqrt(n_predictors)) may cause problems. I am happy to change it if that is helpful.

**Comment by hfrick** [comment]
@topepo can you have a look at this, too? This PR now contains all the changes related to the new `aorsf` engine for censored regression with `rand_forest()`:

- update to the `tunable()` method for an engine-specific argument
- a `fit_xy()` method for random forests: this is so that it can error properly for this engine. I've opened an issue for it to discuss with @bcjaeger (Hi! :wave:) if aorsf could change to enable that https://github.com/ropensci/aorsf/issues/11
- the engine docs

The corresponding PR on censored is https://github.com/tidymodels/censored/pull/211 if you want more context.

File: man/rmd/rand_forest_aorsf.Rmd (hunk 1 above)
Comment (topepo) [reply man/rmd/rand_forest_aorsf.Rmd:11]: No, that's fine. It doesn't get executed and people should know what `n_predictors` represents.

**Review by topepo — APPROVED** [review]

**Comment by hfrick** [comment]
todo for myself: add a note about case weights to the docs

**Comment by hfrick** [comment]
@topepo this is ready to merge 👍 

`fit_xy()` now works with the dev version of aorsf thanks to @bcjaeger changes in aorsf but I'm leaving `fit_xy.rand_forest()` in parsnip until the dev version makes it to CRAN.

**Comment by bcjaeger** [comment]
Thank you! 😃

**Pull request merged**